cli/lib/create-components.js

```diff
@@ -4,9 +4,14 @@ import log from "./log.js";
 import {pascalToKebab} from "./utils.js"
 
 async function* makeComponents(names, template, stylesheet) {
-  for (const name of names) {
+  for (const input of names) {
+    // account for the component being in a sub-directory:
+    const parts = input.split("/");
+    const name = parts.pop();
     const slug = pascalToKebab(name);
-    const directory = `./src/components/${slug}`;
+    const subDirectory = parts.join("/");
+
+    const directory = `./src/components/${subDirectory ? subDirectory + "/" : ""}${slug}`;
```

Comment:
This isn't a deal breaker by any means, but this logic of breaking down the slug and directory is mostly identical across components, locales and pages. You could perhaps move it to utils?

Reply:
Couldn't agree more. This now pulls from a helper called `assembleDirectoryPath`, and returns various useful bits like the final directory, the name, slug, etc. Thanks for the feedback 🙏